lib/puppet/functions/simplib/passgen.rb

```diff
@@ -0,0 +1,127 @@
+# vim: set expandtab ts=2 sw=2:
+Puppet::Functions.create_function(:'simplib::passgen') do
+  dispatch :passgen do
+    param          'String', :param
+    optional_param 'Hash',    :options
+  end
+  def gen_random_pass(length,complexity,complex_only)
+
+    length = length.to_i
+    if length.eql?(0)
+      length = @default_password_length
+    elsif length < 8
+      length = 8
+    end
+
+    passwd = ''
+    begin
+      Timeout::timeout(30) do
+        default_charlist = ("a".."z").to_a + ("A".."Z").to_a + ("0".."9").to_a
+        specific_charlist = nil
+        case complexity
+        when 1
+          specific_charlist = ['@','%','-','_','+','=','~']
```

Comment:
Please add spaces between these elements. It's quite difficult to read like this.